### PR TITLE
chore(connections): replace connections registry and manager with a single interface COMPASS-8469 COMPASS-8470

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch removes two deprecated interfaces (connections repository and manager) and replaces them with a single dependency injectable interface

### CODE REVIEW

File: packages/compass-aggregations/src/components/create-view-modal/create-view-modal.tsx
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/create-view-modal/create-view-modal.tsx:1]: Drive-by to remove usage of a class component

**Review by lerouxb — APPROVED** [review]
